seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/catalog/HiveJDBCCatalog.java

```diff
@@ -0,0 +1,451 @@
+                            .collect(Collectors.toList());
+        }
+        String columnDef =
+                columns.stream()
+                        .map(HiveTypeConvertor::columnToHiveType)
+                        .collect(Collectors.joining(",\n"));
+
+        String ddl =
+                template.replaceAll(
+                                SaveModePlaceHolder.DATABASE.getReplacePlaceHolder(),
+                                tablePath.getDatabaseName())
+                        .replaceAll(
+                                SaveModePlaceHolder.TABLE_NAME.getReplacePlaceHolder(),
+                                tablePath.getTableName())
+                        .replace(SaveModePlaceHolder.ROWTYPE_FIELDS.getPlaceHolder(), columnDef);
+        log.info("EXECUTE DDL SQL is \n {} \n", ddl);
+        try (Statement statement = connection.createStatement()) {
+            statement.execute(ddl);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void dropTable(TablePath tablePath, boolean ignoreIfNotExists)
+            throws TableNotExistException, CatalogException {
+        String query = getDropTableQuery(tablePath, ignoreIfNotExists);
+        try (Statement stmt = connection.createStatement()) {
+            stmt.execute(query);
+        } catch (SQLException e) {
+            throw new CatalogException(e);
+        }
+    }
+
+    @Override
+    public void createDatabase(TablePath tablePath, boolean ignoreIfExists)
+            throws DatabaseAlreadyExistException, CatalogException {
+        String query = getCreateDatabaseQuery(tablePath.getDatabaseName(), ignoreIfExists);
+        try (Statement stmt = connection.createStatement()) {
+            stmt.execute(query);
+        } catch (SQLException e) {
+            throw new CatalogException(
+                    String.format("create database [%s] failed", tablePath.getDatabaseName()), e);
+        }
+    }
+
+    @Override
+    public void dropDatabase(TablePath tablePath, boolean ignoreIfNotExists)
+            throws DatabaseNotExistException, CatalogException {
+        String query = getDropDatabaseQuery(tablePath.getDatabaseName(), ignoreIfNotExists);
+        try (Statement stmt = connection.createStatement()) {
+            stmt.execute(query);
+        } catch (SQLException e) {
+            throw new CatalogException(
+                    String.format("drop database [%s] failed", tablePath.getDatabaseName()), e);
+        }
+    }
+
+    @Override
+    public void truncateTable(TablePath tablePath, boolean ignoreIfNotExists)
+            throws TableNotExistException, CatalogException {
+        throw new UnsupportedOperationException("Does not support truncate table!");
+    }
+
+    @Override
+    public boolean isExistsData(TablePath tablePath) {
+        String tableName = tablePath.getFullName();
+        String sql = String.format("select * from %s limit 1;", tableName);
```

Comment:
`Wouldn't it be better to use show create table`

Reply:
no, this method is to check whether this table has data.
if use `show create table`, can't check has data or not. can only check table exist